xarray/core/formatting.py

```diff
@@ -180,20 +199,33 @@ def format_array_flat(array, max_width):
     array that will fit within max_width characters.
     """
     # every item will take up at least two characters, but we always want to
-    # print at least one item
-    max_possibly_relevant = max(int(np.ceil(max_width / 2.0)), 1)
-    relevant_items = first_n_items(array, max_possibly_relevant)
-    pprint_items = format_items(relevant_items)
-
-    cum_len = np.cumsum([len(s) + 1 for s in pprint_items]) - 1
-    if (max_possibly_relevant < array.size or (cum_len > max_width).any()):
-        end_padding = u' ...'
-        count = max(np.argmax((cum_len + len(end_padding)) > max_width), 1)
-        pprint_items = pprint_items[:count]
+    # print at least first and last items
+    max_possibly_relevant = min(max(array.size, 1),
+                                max(int(np.ceil(max_width / 2.)), 2))
+    relevant_front_items = format_items(
+        first_n_items(array, (max_possibly_relevant + 1) // 2))
+    relevant_back_items = format_items(
+        last_n_items(array, max_possibly_relevant // 2))
+    # interleave relevant front and back items:
+    #     [a, b, c] and [y, z] -> [a, z, b, y, c]
+    relevant_items = sum(zip_longest(relevant_front_items,
+                                     reversed(relevant_back_items)),
+                         ())[:max_possibly_relevant]
+
+    cum_len = np.cumsum([len(s) + 1 for s in relevant_items]) - 1
+    if (max_possibly_relevant < array.size) or (cum_len > max_width).any():
+        padding = u' ... '
+        count = min(array.size,
+                    max(np.argmax(cum_len + len(padding) - 1 > max_width), 2))
     else:
-        end_padding = u''
-
-    pprint_str = u' '.join(pprint_items) + end_padding
+        count = array.size
+        padding = u'' if (count <= 1) else u' '
+
+    num_front = (count + 1) // 2
+    num_back = count - num_front
+    pprint_str = u' '.join(relevant_front_items[:num_front]) + \
```

Comment:
nit: can you use parentheses here instead of `\` for line continuation?

Reply:
Done.